ReadableStream TypeScript Compiler Error

Hey there, we're running into the following TypeScript compiler error:

https://github.com/seek-oss/skuba/actions/runs/10189049455/job/28186365681?pr=1622#step:9:36

```console
Error: tsc      │ node_modules/.pnpm/memfs@4.11.0/node_modules/memfs/lib/node/types/misc.d.ts(109,61): error TS2552: Cannot find name 'ReadableStream'. Did you mean 'IReadStream'?
```

This only exists with the `dom` lib option which is not true for backend only projects.

Would you be okay with the approach the aws sdk v3 took? https://github.com/aws/aws-sdk-js-v3/pull/3889/files

--- a/src/node/types/misc.ts
+++ b/src/node/types/misc.ts
@@ -122,6 +122,15 @@ export interface IFSWatcher extends EventEmitter {
   close(): void;
 }
 
+/**
+ * Declare ReadableStream in case dom.d.ts is not added to the tsconfig lib causing
+ * ReadableStream interface is not defined. For developers with dom.d.ts added,
+ * the ReadableStream interface will be merged correctly.
+ */
+declare global {
+  export interface ReadableStream {}
+}
+
 export interface IFileHandle {
   fd: number;
   appendFile(data: TData, options?: IAppendFileOptions | string): Promise<void>;

